src/renderer/components/dock/monaco-model-manager.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+import { action, makeObservable } from "mobx";
+import {monaco} from "react-monaco-editor";
+
+export type TabId = string;
+
+interface ModelEntry {
+    id?: TabId;
+    modelUri?: monaco.Uri;
+    lang?: string;
+  }
+  
+export interface ModelsState {
+    models: ModelEntry[];
+  } 
+  
+
+export class MonacoModelsManager implements ModelsState {
+    private modelsArray = new Array<ModelEntry>();
+
+    constructor() {
+      makeObservable(this);
+    }
+    
+    get models(){
+      return this.modelsArray;
+    }
+
+    set models(models: ModelEntry[]) {
+      this.modelsArray = models;
+    }
+
+    @action
+    addModel(tabId: string, { value = "", lang = "yaml" } = {}): monaco.editor.ITextModel {
+      let model;
+      const uri = this.getUri(tabId);
+
+      if(uri) {
+        model = monaco.editor.createModel(value, lang, uri); 
+      } else {
+        model = monaco.editor.createModel(value, lang); 
+      }
+        
+      !uri && (this.models = this.models.concat({ id: tabId, modelUri: model.uri, lang}));
+
+      return model;
+    }
+
+    getModel(tabId: string): monaco.editor.ITextModel {
+      return monaco.editor.getModel(this.getUri(tabId));
+    }
+
+    getUri(tabId: string): monaco.Uri {
+      return this.models.find( model => model.id == tabId)?.modelUri;
+    }
+
+    @action
+    removeModel(tabId: string) {
+      const uri = this.getUri(tabId);
+
+      this.models = this.models.filter(v => v.id != tabId);
+      monaco.editor.getModel(uri).dispose();
+    }
```

Comment:
If this gets called before the reactions and `componentDidMount` methods gets called the I would assume that `getModel` would return `undefined`.